src/mongocxx/collection.cpp

```diff
@@ -624,9 +624,15 @@ bsoncxx::document::value collection::create_index(bsoncxx::document::view keys,
         throw exception::operation(std::make_tuple(error.message, error.code));
     }
 
-    // TODO: return the response from the server, this is not possible now due to the way
-    // libmongoc works.
-    return bsoncxx::document::value{bsoncxx::document::view{}};
+    if (options.name()) {
+        return bsoncxx::builder::stream::document{} 
```

Comment:
i guess the way i suggested always calls collection_keys_to_index_string even if it is unneeded. in that case, I think a few extra lines justify the performance benefit. So maybe better to keep as is